mewarpx/tests/test_emission.py

```diff
@@ -85,6 +85,84 @@ def test_thermionic_emission():
     assert np.allclose(net_rho_grid, ref_rho_grid)
 
 
+def test_thermionic_emission_with_Schottky():
```

Comment:
It would be good to verify injection statistics here, because otherwise we won't be certain the weights update in python is working correctly for the record.

Reply:
That is true, although the test as is compares the charge density at the end of the simulation to a reference solution so that does capture the upweighting of particles in a less direct way.

Reply:
Maybe we can add a card to add such a test so this PR isn't held up by it?

Reply:
Sure, I'll create a separate card. My point is that if the weights are updated on the C++ side but not fed back to python correctly, recorded injection statistics can be wrong even as the charge density comparison is correct.

Reply:
Oh I see what you mean. Yeah the flux diagnostic correctness is not tested and definitely should be (although I can confirm from the vacuum diode tests that it does currently work).